libkernelflinger/vars.c

```diff
@@ -382,12 +391,16 @@ EFI_STATUS set_current_state(enum device_state state)
 	if (!is_live_boot()) {
 #ifdef SECURE_STORAGE_RPMB
 		ret = write_rpmb_device_state(stored_state);
-#else
-		ret = write_device_state_efi(stored_state);
-#endif
+#else /* SECURE_STORAGE_RPMB */
+#ifdef USE_TPM
+		if (!is_platform_secure_boot_enabled() ||
+				(ret = write_device_state_tpm2(stored_state)) == EFI_NOT_FOUND)
```

Comment:
Indentation is wrong.  Also this is a bad habit to assign value in a test condition like this.